rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -529,13 +542,20 @@ func (r *Layer2Relayer) ProcessPendingBundles() {
 
 func (r *Layer2Relayer) finalizeBundle(bundle *orm.Bundle, withProof bool) error {
 	// Check batch status before sending `finalizeBundle` tx.
-	if r.cfg.ChainMonitor.Enabled {
-		for batchIndex := bundle.StartBatchIndex; batchIndex <= bundle.EndBatchIndex; batchIndex++ {
-			tmpBatch, getErr := r.batchOrm.GetBatchByIndex(r.ctx, batchIndex)
-			if getErr != nil {
-				log.Error("failed to get batch by index", "batch index", batchIndex, "error", getErr)
-				return getErr
-			}
+	for batchIndex := bundle.StartBatchIndex; batchIndex <= bundle.EndBatchIndex; batchIndex++ {
```

Comment:
add current bunlde codec version is larger than the preceding one

Reply:
makes sense. added in [873d9d9](https://github.com/scroll-tech/scroll/pull/1562/commits/873d9d9980615acd53887731fd46694ade9ec56e).